File: crons-sample/src/main/java/com/raxdenstudios/sample/MainActivity.java
package com.raxdenstudios.sample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.raxdenstudios.cron.CronHandler;
import com.raxdenstudios.cron.model.Cron;

import java.util.Calendar;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.observers.DisposableCompletableObserver;
import io.reactivex.schedulers.Schedulers;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();

    private Disposable mDisposable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        long now = Calendar.getInstance().getTimeInMillis();

        final Cron cron = new Cron.Builder(1)
                .triggerAtTime(now + 10000)
                .interval(10000)
                .create();

        final Cron cron2 = new Cron.Builder(2)
                .triggerAtTime(now + 30000)
                .interval(30000)
                .create();

        final Cron cron3 = new Cron.Builder(3)
                .triggerAtTime(now + 60000)
                .interval(60000)
                .create();

        CronHandler handler = new CronHandler(this);

        mDisposable = handler.start(cron)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeWith(new DisposableCompletableObserver() {
                    @Override
                    public void onComplete() {
                        Log.d(TAG,"onComplete!");
                    }

                    @Override
                    public void onError(Throwable e) {
                        Log.e(TAG, e.getMessage(), e);
                    }
                });
    }

    @Override
    protected void onDestroy() {
        if (mDisposable != null && !mDisposable.isDisposed()) {
            mDisposable.dispose();
        }
        super.onDestroy();
    }
}

File: crons-sample/src/main/java/com/raxdenstudios/sample/MainActivity.kt
package com.raxdenstudios.sample

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.raxdenstudios.cron.CronHandler
import com.raxdenstudios.cron.model.Cron
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.observers.DisposableCompletableObserver
import io.reactivex.schedulers.Schedulers
import java.util.*

class MainActivity : AppCompatActivity() {

    companion object {
        private val TAG = MainActivity::class.java.simpleName
    }

    private val compositeDisposable: CompositeDisposable = CompositeDisposable()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val now = Calendar.getInstance().timeInMillis

        val cron = Cron.Builder(1).triggerAtTime(now + 10000).interval(10000).create()
        val cron2 = Cron.Builder(2).triggerAtTime(now + 30000).create()
        val cron3 = Cron.Builder(2).triggerAtTime(now + 60000).create()

        compositeDisposable.add(CronHandler(this).start(cron)
                .andThen(CronHandler(this).start(cron2))
                .andThen(CronHandler(this).start(cron3))
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeWith(object : DisposableCompletableObserver() {
                    override fun onComplete() {
                        Log.d(TAG, "onComplete!")
                    }

                    override fun onError(e: Throwable) {
                        Log.e(TAG, e.message, e)
                    }
                }))
    }

    override fun onDestroy() {
        if (!compositeDisposable.isDisposed)
            compositeDisposable.dispose()

        super.onDestroy()
    }
}